Update to OpenJDK 20 and FreeDesktop runtime 22.08

tsmock opened the pull request:
Update to OpenJDK 19. This also updates maven to 3.9.0, gradle to 8.0.1, and ant to 1.10.13.

This will fix #13 .

This also adds update checkers for minor revisions (using the [Adoptium API](https://api.adoptium.net/q/swagger-ui/)).

tsmock committed:
Update to OpenJDK 18

Signed-off-by: Taylor Smock <tsmock@fb.com>

tsmock committed:
Initial auto update

Signed-off-by: Taylor Smock <tsmock@fb.com>

tsmock commented:
I've updated to jdk-18.0.1.1+2. Note that I've also added update checkers for ant, gradle, and maven (see [flathub/flatpak-external-data-checker](https://github.com/flathub/flatpak-external-data-checker)). I did not add an update checker for the jdk, as `curl https://api.github.com/repos/openjdk/jdk18u/tags | jq '.[0]'` gives [jdk-18+37](https://github.com/openjdk/jdk18u/releases/tag/jdk-18%2B37), which is pre-GA for Java 18.

muelli commented:
any takers? @mbooth101?

1player commented:
Welp.. no update from the maintainer. I guess it's also worth tagging it 22.08 now.

tsmock commented:
I'll probably update this PR for JDK 19 after 2022-09-20 (JDK 19 GA date, according to https://openjdk.org/projects/jdk/19/ ). But that depends upon the bootstrap JDK being updated (this repo has been using https://fedorapeople.org/~mbooth/bootstrap_jdk/, which I believe is maintained by @mbooth101 ), so I might have to do some stuff to get the bootstrap JDK set up. I'll probably switch my PR to use adoptium instead.

tsmock committed:
Update to OpenJDK 19

Notes:
* Temurin is used for bootstrapping the JDK
* Use Temurin for x-checker-data for non-major releases (19.x.y)
* Manual action will still be required for major version updates
  such as JDK19 -> JDK20.
* Manual action is still required for updating the symbolic link
  on post_install (19.0.0 -> new revision)

Signed-off-by: Taylor Smock <tsmock@fb.com>

SailReal commented:
https://github.com/flathub/org.cryptomator.Cryptomator requires Java 19 to update to version [1.6.16](https://github.com/cryptomator/cryptomator/releases/tag/1.6.16)

Would be quiet awesome if this PR could be shipped :pray:

tsmock committed:
Update to 22.08 org.freedesktop.Platform, update to latest package versions

Signed-off-by: Taylor Smock <tsmock@meta.com>

tsmock committed:
Use bash * to avoid having to update changing version directory

Signed-off-by: Taylor Smock <tsmock@meta.com>

tsmock committed:
Work around two certificates using the same CN

Signed-off-by: Taylor Smock <tsmock@meta.com>

tsmock commented:
@mbooth101 : Are you still maintaining org.freedesktop.Sdk.Extension.openjdk? If not, who is?

tsmock committed:
Update to Java 20

Also update
* ant to 1.10.13
* maven to 3.9.1
* gradle to 8.0.2

tsmock committed:
Modify gradle version-query and update it to 8.1

The previous version-query did not work for the initial versions of each
minor release (X.Y.0 would always fail).

Signed-off-by: Taylor Smock <tsmock@meta.com>

muelli commented on org.freedesktop.Sdk.Extension.openjdk.appdata.xml:
```diff
@@ -15,4 +15,7 @@
   <developer_name>Mat Booth</developer_name>
   <update_contact>mat.booth@gmail.com</update_contact>
+  <releases>
+    <release version="19.0.2+7" date="2023-02-21"/>
```
hrmpf.
I thought there was a way of "suggesting" an edit by performing it directly. But I can't find it now.
It currently says "19", but you're building "20", no?

tsmock replied:
Thanks for pointing that out. I forgot to do a search and replace for 19.

tsmock replied:
And fixed -- that was automatically added when I used `org.flathub.flatpak-external-data-checker` to run an update. I've dropped the `<releases>` node since there never was release with Java 19.

tsmock commented:
@barthalion : Pinging since this repo doesn't seem to have an active maintainer.

muelli commented:
yeah, would be great to have somebody maintain this extension.

barthalion commented:
@tsmock I have sent you an invite to collaborators.

ghazyami commented:
Next version of https://github.com/flathub/com.gluonhq.SceneBuilder requires OpenJDK 20, would love if this PR is merged.

Thanks!

tsmock commented:
I'm planning on merging this Friday, when I have a bit of time to do one last update (a Java 20 point release was done after my last update).

tsmock committed:
Update to Java 20.0.1

Also update Maven to 3.9.2 and Gradle to 8.1.1.

Signed-off-by: Taylor Smock <smocktaylor@gmail.com>

tsmock commented:
OK. I've pushed a new branch with the changes ([branch/22.08](https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk/tree/branch/22.08)).

@barthalion : I think someone with admin access has to change the default branch (I updated the base runtime from 21.08 to 22.08). Unless this repo is out of [spec](https://github.com/flathub/flathub/wiki/App-Requirements#branches-and-runtimes), in which case I'll need to push a `stable` branch.

barthalion commented:
I have updated the default branch.
